legal-api/src/legal_api/models/business.py

```diff
@@ -397,11 +397,17 @@ def is_firm(self):
     @property
     def good_standing(self):
         """Return true if in good standing, otherwise false."""
+        from legal_api.services import flags  # pylint: disable=import-outside-toplevel
+
         # A firm is always in good standing
         if self.is_firm:
             return True
         # Date of last AR or founding date if they haven't yet filed one
         last_ar_date = self.last_ar_date or self.founding_date
+        # When involuntary dissolution feature flag is on, check transition filing
+        if flags.is_on('enable_involuntary_dissolution'):
+            if self._has_no_transition_filed_after_restoration():
+                return False
```

Comment:
Can you move the block up a few lines? Because this cuts off the logic that checks the last AR date.

Reply:
Thank you for pointing it out, addressed it. 😄